### ISSUE TITLE
How can I update my Solr 8 schema.xml to include Sunspot's fields?

### ISSUE THREAD

**Issue by gr8bit** [sunspot/sunspot#983]
Hello all, I would like to use Sunspot with Solr 8 but cannot find enough information for me to understand which fields, dynamicFields etc. have to be added to the default schema.xml. **I already found the solr 7 schema.xml config** in this repo but that didn't exactly help my understanding and seems to be outdated a bit, too: we're using Sunspot with Solr 7.7 in production and it has only half of the extra (dynamic)fields of that config and works perfectly.

So in short: I want to be able to add Sunspot's required Solr configuration to its [current default schema](https://github.com/apache/lucene-solr/blob/releases/lucene-solr/8.6.0/solr/server/solr/configsets/_default/conf/managed-schema).

Does anybody of you have hints on where to find the required configuration in the Sunspot code?

**Comment by mlh758** [comment]
These schemas aren't Sunspot specific, they are part of Apache Solr itself. Some resources:

https://lucene.apache.org/solr/guide/8_6/solr-field-types.html
https://lucene.apache.org/solr/guide/8_0/overview-of-documents-fields-and-schema-design.html

**Comment by gr8bit** [comment]
I think I know, but Sunspot needs some extra config like "class_name" or "type" fields for example. I would like to exactly know which so I can add those to my Solr scheme. With the (above linked) default schema, Sunspot won't work.

I think Sunspot's Solr-DSL is one of the greatest Rails-addons out there and I've totally come to love the combination of Sunspot/Solr for its simplicity and its power! <3 That why I'd like to keep that config part up to date.

**Comment by mlh758** [comment]
```
<fieldType name="string" class="solr.StrField" omitNorms="true"/>
<dynamicField name="*_texts" stored="true" type="text" multiValued="true" indexed="true"/>
<field name="class_name" stored="false" type="string" multiValued="false" indexed="true"/>
```

I just pulled some examples out. The `class` is Specific to Solr. The names are used by Sunspot when it matches fields up dynamically as you add stuff to your index. For example in one of our cores we have this schema present:

![image](https://user-images.githubusercontent.com/4331955/87550891-2e63ff00-c675-11ea-808f-ab87519b01d5.png)

Notice how it matches up with the dynamicField definition. What errors specifically are you getting? If you were wanting to use this with Solr 8 you would start by removing all the [trie](https://github.com/sunspot/sunspot/blob/master/examples/solr7_core/conf/schema.xml#L84) fields from the available types and any reference to them below. I don't _think_ the sunspot gem will try to access them unless you use them but I haven't tried so I don't know for sure.

**Comment by gr8bit** [comment]
Thanks a lot for your time! With the Solr 8 default config, I'm getting the error I later added in my post above. Note that I did not replace the Solr 8 default config with the config from this repo - I want to keep all the (new) defaults and only add Sunspot's required fields.

Do you have a field name="type" in your xml?

**Comment by mlh758** [comment]
The example config in this repo is very nearly a copy/paste of the config I use in one of my projects. I PR'd it in because we've had a lot of questions in the past about how to use solr 7 with this gem. You probably need most of the field and dynamic field definitions that are in the solr schema. They do after all have the comment about being used by Sunspot :)

The error about type not being defined corresponds to [this line](https://github.com/sunspot/sunspot/blob/master/examples/solr7_core/conf/schema.xml#L133).

**Comment by gr8bit** [comment]
I started going through the "used by Sunspot" definitions line per line and added them to the Solr default config first. It then occurred to me I don't even have a lot of them in my Solr 7.7 config (which runs fine). So I'm wondering which of the fields marked as "used by Sunspot" are _actually_ used by Sunspot any more. It for sure aren't all of those fields.

Do you have any insight from where in this repo's code I could compile a "used by Sunspot"-field list myself?

**Comment by mlh758** [comment]
Some of them are here: https://github.com/sunspot/sunspot/blob/master/sunspot/lib/sunspot/dsl/fields.rb
More directly here: https://github.com/sunspot/sunspot/blob/master/sunspot/lib/sunspot/schema.rb#L16

But what is your aversion to having all the fields that Sunspot uses in your schema? For example, you can certainly remove the `_b` dynamic field, and the gem will work, until you try to index a boolean. Why set yourself up for frustration later? It seems like it would be easier to add all the fields, and then remove the stuff you know is not available in Solr 8 like Trie fields.

**Comment by gr8bit** [comment]
Thanks! I don't want to remove any fields from the Solr 8 default schema and `_b` is already in that default - I just want to collect the fields Sunspot needs _added_ to the default config. I'd like to add explicitly add these fields to any (possibly already customized) schema rather than having to overwrite it. Plus I like to learn ;)

**Comment by gr8bit** [comment]
@mlh758 having dug through the code a while, I now understand why you questioned my undertaking. The types from the default schema are defined quite differently than the (older) core-7 schema from this repo - they were adapted to suit the needs. I'll still compiling a list, but I will go with the existing schema that worked for 7.7 so well.
Thank you a lot for the explanation!

**Comment by mlh758** [comment]
I'm going to close this for now then, I can reopen the issue if you end up needing more help. Good luck!
